# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/inkscape/hotfix-poppler.patch
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- inkscape-1.4.3_2025-12-25_0d15f75042/src/extension/internal/pdfinput/svg-builder.cpp.vanilla	2026-01-10 19:32:10.059226801 +0100
+++ inkscape-1.4.3_2025-12-25_0d15f75042/src/extension/internal/pdfinput/svg-builder.cpp	2026-01-10 19:36:58.969575370 +0100
@@ -27,6 +27,7 @@
 
 #ifdef HAVE_POPPLER
 
+#include <poppler/goo/gmem.h>
 #include <poppler/Function.h>
 #include <poppler/GfxFont.h>
 #include <poppler/GfxState.h>
@@ -2093,7 +2094,11 @@
         } else {
             image_stream = new ImageStream(str, width, 1, 1);
         }
+#if POPPLER_CHECK_VERSION(26, 1, 0)
+        image_stream->rewind();
+#else
         image_stream->reset();
+#endif
 
         // Convert grayscale values
         unsigned char *buffer = new unsigned char[width];
@@ -2119,7 +2120,11 @@
         image_stream = new ImageStream(str, width,
                                        color_map->getNumPixelComps(),
                                        color_map->getBits());
+#if POPPLER_CHECK_VERSION(26, 1, 0)
+        image_stream->rewind();
+#else
         image_stream->reset();
+#endif
 
         // Convert RGB values
         unsigned int *buffer = new unsigned int[width];
--- inkscape-1.4.3_2025-12-25_0d15f75042/src/extension/internal/pdfinput/pdf-input.cpp.vanilla	2026-01-10 19:31:24.971394886 +0100
+++ inkscape-1.4.3_2025-12-25_0d15f75042/src/extension/internal/pdfinput/pdf-input.cpp	2026-01-10 19:37:37.750814254 +0100
@@ -19,6 +19,7 @@
 #include "pdf-input.h"
 
 #ifdef HAVE_POPPLER
+#include <poppler/goo/gmem.h>
 #include <poppler/Catalog.h>
 #include <poppler/ErrorCodes.h>
 #include <poppler/FontInfo.h>
